### PR TITLE
fix for order sms and edit profile email issue

### PR DESCRIPTION
**Pull request by amarpanday**
fix for order sms and edit profile email issue.

### CODE REVIEW

File: sendinblue.php [hunk 1]
```diff
@@ -187,11 +186,16 @@ public function callhookRegister()
             $this->months = Tools::getValue('months');
             $this->years = Tools::getValue('years');
             $birthday = ($this->years . '-' . $this->months . '-' . $this->days);
-            
+
+            echo '<script type="text/javascript">
```
Comment (Mekki78) [inline sendinblue.php]: Javascript code need to be placed in a js file and located in yourmodule/views/js/yourjsfile.js also you should use the addJs function to add it in your module.  
`$this->context->controller->addJs($this->_path . 'js/newsletter.js');`

File: sendinblue.php [hunk 2]
```diff
@@ -32,16 +32,19 @@
...
 }
 
-include_once (dirname(__FILE__) . '/sendinblueResources.php');
 if (version_compare(_PS_VERSION_, '1.4.5', '<')) {
+    require(_PS_CLASS_DIR_. '/../modules/sendinblue/Mailin.php');
+    include_once (_PS_CLASS_DIR_. '/../modules/sendinblue/sendinblueResources.php');
     $sendinblue_resources = new SendinblueResources();
     if ($sendinblue_resources->checkConditionOlderVersion()) {
-        include (dirname(__FILE__) . '/config.php');
+        include (_PS_CLASS_DIR_. '/../modules/sendinblue/config.php');
```
Comment (Mekki78) [inline sendinblue.php]: include (dirname(__FILE__) . '/config.php');
this was the good utilisation to catch your file you want to include. Please use dirname(__FILE__).

File: sendinblue.php [hunk 3]
```diff
@@ -32,16 +32,19 @@
     include_once (_PS_CLASS_DIR_ . '/../classes/Customer.php');
 }
 
-include_once (dirname(__FILE__) . '/sendinblueResources.php');
 if (version_compare(_PS_VERSION_, '1.4.5', '<')) {
+    require(_PS_CLASS_DIR_. '/../modules/sendinblue/Mailin.php');
```
Comment (Mekki78) [inline sendinblue.php]: include (dirname(__FILE__) . '/Mailin.php');
this was the good utilisation to catch your file you want to include. Please use dirname(__FILE__).

File: sendinblue.php [hunk 4]
```diff
@@ -32,16 +32,19 @@
...
+    include_once (_PS_CLASS_DIR_. '/../modules/sendinblue/sendinblueResources.php');
```
Comment (Mekki78) [inline sendinblue.php]: include (dirname(__FILE__) . '/sendinblueResources.php');
this was the good utilisation to catch your file you want to include. Please use dirname(__FILE__).

File: sendinblue.php [hunk 5]
```diff
@@ -32,16 +32,19 @@
     include_once (_PS_CLASS_DIR_ . '/../classes/Customer.php');
```
Comment (Mekki78) [inline sendinblue.php]: include (dirname(__FILE__) . '/classes/Customer.php');
this was the good utilisation to catch your file you want to include. Please use dirname(__FILE__).

File: Mailin.php [hunk 6]
```diff
@@ -0,0 +1,1186 @@
...
+ * @copyright 2007-2014 PrestaShop SA
+ * @license   http://opensource.org/licenses/afl-3.0.php Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+class Mailin
+{
+    public $api_key;
+    public $base_url;
+    public $curl_opts = array();
```
Comment (Mekki78) [inline Mailin.php]: Unused variable `$curl_opts`

File: Mailin.php [hunk 7]
```diff
@@ -0,0 +1,1186 @@
...
+        }
+        curl_setopt($ch, CURLOPT_HTTPHEADER, array($auth_header, $content_header));
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_HEADER, 0);
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $input);
+        $data = curl_exec($ch);
+        if (curl_errno($ch)) {
+            echo 'Curl error: ' . curl_error($ch). '\n';
```
Comment (Mekki78) [inline Mailin.php]: You need to manage errors and display them correctly. 
If something wrong happen your client will see something like this (screenshot)

download 16

You can show the error only if the dev mode is enabled for example:

/* Debug only */
if (!defined('_PS_MODE_DEV_')) {
    // your code
}

File: sendinblue.php (hunk 5 above)
Comment (amarpanday) [inline sendinblue.php]: dirname(FILE)  not working  for prestashop version 1.4.x, in module list page .
and to make it working for 1.4.x we are using _PS_CLASS_DIR_ for other version we are using as you asked.

File: sendinblue.php (hunk 3 above)
Comment (amarpanday) [inline sendinblue.php]: dirname(FILE)  not working  for prestashop version 1.4.x, in module list page .
and to make it working for 1.4.x we are using _PS_CLASS_DIR_ for other version we are using as you asked.

File: sendinblue.php (hunk 2 above)
Comment (amarpanday) [inline sendinblue.php]: dirname(FILE)  not working  for prestashop version 1.4.x, in module list page .
and to make it working for 1.4.x we are using _PS_CLASS_DIR_ for other version we are using as you asked.

File: sendinblue.php (hunk 1 above)
Comment (amarpanday) [inline sendinblue.php]: we done accordingly

File: Mailin.php (hunk 6 above)
Comment (amarpanday) [inline Mailin.php]: we done accordingly

File: Mailin.php (hunk 7 above)
Comment (amarpanday) [inline Mailin.php]: we done accordingly

**Pull request merged**